### ISSUE TITLE
Flaky ZX outcome probability test: unseeded 10-shot estimate can fail on valid samples

### ISSUE THREAD

**Issue by QingyunQian** [tensorcircuit/tensorcircuit-ng#147]
## Issue Description

`tests/test_zx.py::test_zx_outcome_probability[jaxb]` can fail on valid random samples from unchanged upstream. Its noisy case uses an unseeded `StabilizerTCircuit`, draws only 10 error realizations with probability 0.1, and checks their mean against 0.1 with `atol=0.3, rtol=0.0`.

This can intermittently fail the full test suite even when the sampler behaves correctly. I reproduced it on upstream commit `75f8961195a973c9c8c48a6d798d781ab1b9fdd3`.

Relevant code: [test and assertion](https://github.com/tensorcircuit/tensorcircuit-ng/blob/75f8961195a973c9c8c48a6d798d781ab1b9fdd3/tests/test_zx.py#L149-L186), [random default seed](https://github.com/tensorcircuit/tensorcircuit-ng/blob/75f8961195a973c9c8c48a6d798d781ab1b9fdd3/tensorcircuit/zx/stabilizertcircuit.py#L111-L133).

## How to Reproduce

Run this on the CPU backend (`JAX_PLATFORMS=cpu`). An explicit seed makes the otherwise intermittent failure deterministic in the environment below:

```python
import numpy as np
import jax.numpy as jnp
from tensorcircuit.zx.stabilizertcircuit import StabilizerTCircuit

circuit = StabilizerTCircuit(1, seed=156)
circuit.x_error(0, 0.1)
circuit.measure_instruction(0)
p1 = circuit.outcome_probability(jnp.array([1]), shots=10)

print(np.asarray(p1))
print(np.mean(p1))
# The same assertion as the existing test:
np.testing.assert_allclose(np.mean(p1), 0.1, atol=0.3, rtol=0.0)
```

### Error Output

```text
[0. 0. 1. 1. 0. 0. 1. 0. 1. 0.]
0.4

Not equal to tolerance rtol=0, atol=0.3
Mismatched elements: 1 / 1 (100%)
Max absolute difference among violations: 0.30000001
Max relative difference among violations: 3.00000006
 ACTUAL: array(0.4, dtype=float32)
 DESIRED: array(0.1)
```

The float32 mean is approximately `0.4000000059604645`, just beyond the comparison boundary. This is the same failure observed in an unseeded full-suite run. An isolated rerun of the unmodified test passed, so a passing rerun alone does not establish stability.

There is also a statistical issue independent of that rounding boundary: replacing the seed with `25` yields five ones out of ten (mean 0.5) and fails the assertion too. A small tolerance adjustment would therefore not remove the underlying flakiness.

## Estimated Failure Probability and Suggested Direction

Under the intended independent Bernoulli model, the error count is `K ~ Binomial(10, 0.1)`. With the observed float32 comparison, counts of four or more fail:

- `P(K = 4)` is about **1.1160%**.
- `P(K >= 5)` is about **0.1635%**.
- Combined, `P(K >= 4)` is about **1.2795% per execution of this test** in these conditions.

For independent executions, even 50 consecutive passes have probability about 52.5%; 100 consecutive passes have probability about 27.6%. Many successful previous runs are therefore consistent with this intermittent failure. These are model-based probabilities for this one assertion, not measured whole-suite failure rates or a claim that every environment has identical float32 behavior.

It may help to separate a reproducible small-sample shape/range check from a distribution check with a larger sample count and a statistically justified tolerance. Fixing only the seed or slightly increasing the tolerance would leave the underlying statistical criterion unexplained. This report concerns test reliability; the sampled outcomes shown above are valid under the intended noise model.

## Environment Context

Upstream source commit: `75f8961195a973c9c8c48a6d798d781ab1b9fdd3`; CPU execution with `JAX_PLATFORMS=cpu`.

Relevant `tc.about()` output and `tc.__version__`:

```text
OS info: Linux-6.17.0-35-generic-x86_64-with-glibc2.39
Python version: 3.12.3
Numpy version: 2.2.6
Scipy version: 1.17.1
TensorNetwork version: 0.5.1
TensorFlow version: 2.20.0
Jax version: 0.7.0
JaxLib version: 0.7.0
PyTorch version: 2.11.0+cpu
tc.__version__: 1.9.1
pytest: 7.4.4
pytest-xdist: 3.8.0
pyzx-param: 0.9.3
```
